### PR TITLE
Add or suppress missing parameter type declarations

### PR DESCRIPTION
**Pull request by othercorey**

### CODE REVIEW

File: src/Cache/CacheEngine.php [hunk 1]
```diff
@@ -97,7 +97,7 @@ public function init(array $config = []): bool
      * @return void
      * @throws \Cake\Cache\InvalidArgumentException When the key is not valid.
      */
-    protected function ensureValidKey($key): void
+    protected function ensureValidKey(string $key): void
```
Comment (ADmad) [inline src/Cache/CacheEngine.php]: This needs to be `mixed` as it's actually a validation function.

Comment (othercorey) [reply src/Cache/CacheEngine.php]: You're right. I was just looking at this error.

File: src/Cache/CacheEngine.php [hunk 2]
```diff
@@ -107,12 +107,12 @@ protected function ensureValidKey($key): void
...
      * Ensure the validity of the argument type and cache keys.
      *
-     * @param iterable $iterable The iterable to check.
+     * @param mixed $iterable The iterable to check.
      * @param string $check Whether to check keys or values.
      * @return void
      * @throws \Cake\Cache\InvalidArgumentException
      */
-    protected function ensureValidType($iterable, string $check = self::CHECK_VALUE): void
+    protected function ensureValidType(mixed $iterable, string $check = self::CHECK_VALUE): void
```
Comment (ADmad) [inline src/Cache/CacheEngine.php]: This needs to stay as `iterable` as it's called from `*Multiple()` methods which already have `iterable` type. The `is_iterable()` check below can be dropped.

Comment (othercorey) [reply src/Cache/CacheEngine.php]: Done. Will throw a TypeError for non-iterable arguments and InvalidArgumentExceptions on invalid key/value types.

File: src/Log/Engine/ArrayLog.php [hunk 3]
```diff
@@ -57,6 +57,7 @@ class ArrayLog extends BaseLog
      * @param array $context Additional information about the logged message
      * @return void success of write.
      * @see \Cake\Log\Log::$_levels
+     * @phpcsSuppress SlevomatCodingStandard.TypeHints.ParameterTypeHint.MissingNativeTypeHint
```
Comment (markstory) [inline src/Log/Engine/ArrayLog.php:60]: These will show up in the API docs. We might want to filter these attributes out when building the API docs if we continue to use them.  Another way to ignore these errors would be in the configuration files.

Comment (othercorey) [reply src/Log/Engine/ArrayLog.php:60]: The new api docs should show only a specific list of attributes.

**Pull request merged**